src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -3287,7 +3287,26 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                     {
                         allPathsString.push_back(quotesChar);
                     }
-                    allPathsString.append(fullPath);
+                    if (isWSL)
+                    {
+                        // Fix quoted path for WSL
+                        // Single quote is allowed on the Win32 subsystem and must be processed
+                        // when we are quoting the path with single quotes (on WSL).
+                        // Note that we assume that all paths are quoted for WSL.
+                        size_t pos;
+                        auto fullPathView = std::wstring_view(fullPath);
+                        while ((pos = fullPathView.find(L"'")) != std::wstring_view::npos)
+                        {
+                            allPathsString.append(fullPathView.begin(), fullPathView.begin() + pos);
+                            allPathsString.append(L"'\"'\"'");
+                            fullPathView.remove_prefix(pos + 1);
+                        }
+                        allPathsString.append(fullPathView);

```

Comment:
There are two things that I believe must be changed:
* If `quotesNeeded` is false, then we must only insert `"'"` and not `'"'"'`, right?
* `find()` has an overload for single character searches that is much faster. So instead of `L"'"` I think we should use `L'\''`.

Afterwards, I think we can merge this. There's another thing though that's less important:
* As per our usual code style it would be better to write this:
  ```cpp
  std::wstring_view fullPathView{ fullPath };
  ```

Reply:
## 1. `quotesNeeded` handling

> * If `quotesNeeded` is false, then we must only insert `"'"` and not `'"'"'`, right?

Seems valid at first glance but...
```cxx
const auto quotesNeeded = isWSL || fullPath.find(L' ') != std::wstring::npos;
```
If `isWSL` is `true`, `quotesNeeded` is always `true` (and I felt I needed to note that).  That's why I inserted:
```cxx
// Note that we assume that all paths are quoted for WSL.
```
in the comment.

If that's not convincing enough, I'll consider either:
*    improving the comment (any ideas?) or
*    handle `quotesNeeded` and `quotesChar` in the `else` block of `if (isWSL)` to completely separate the logic between WSL and non-WSL paths.

## 2. `find()` overload

> * `find()` has an overload for single character searches that is much faster. So instead of `L"'"` I think we should use `L'\''`.

I completely missed that (and will follow your review).

## 3. Code style

> * As per our usual code style it would be better to write this:
>   ```cxx
>   std::wstring_view fullPathView{ fullPath };
>   ```

I'll change that, too.

Reply:
Thank you for pointing out my mistake with the `quotesNeeded` flag!